### PR TITLE
fix: retrive only the last "next" draft release id

### PR DESCRIPTION
**Pull request by lemeurherve**
Fixes https://github.com/jenkins-infra/pipeline-library/issues/379

### CODE REVIEW

**Review by dduportal — DISMISSED** [review]

File: vars/buildDockerAndPublishImage.groovy [hunk 1]
```diff
@@ -196,7 +196,7 @@ def call(String imageName, Map userConfig=[:]) {
               fi
               '''
 
-              final String release = sh(returnStdout: true, script: 'gh api ${GH_RELEASES_API_URI} | jq -e -r \'.[] | select(.draft == true and .name == "next") | .id\'').trim()
+              final String release = sh(returnStdout: true, script: 'gh api ${GH_RELEASES_API_URI} | jq -e -r \'.[] | select(.draft == true and .name == "next") | .id | sort_by(id) | reverse | first\'').trim()
```
Comment (lemeurherve) [inline vars/buildDockerAndPublishImage.groovy:199]:
```suggestion
              final String release = sh(returnStdout: true, script: 'gh api ${GH_RELEASES_API_URI} | jq -e -r '. | max_by(.id) | select(.draft == true and .name == "next").id\'').trim()
```

File: vars/buildDockerAndPublishImage.groovy [hunk 2]
```diff
@@ -196,7 +196,7 @@ def call(String imageName, Map userConfig=[:]) {
               fi
               '''
 
-              final String release = sh(returnStdout: true, script: 'gh api ${GH_RELEASES_API_URI} | jq -e -r \'.[] | select(.draft == true and .name == "next") | .id\'').trim()
+              final String release = sh(returnStdout: true, script: 'gh api ${GH_RELEASES_API_URI} | jq -e -r '. | max_by(.id) | select(.draft == true and .name == "next").id\'').trim()
```
Comment (lemeurherve) [inline vars/buildDockerAndPublishImage.groovy:199]:
```suggestion
              final String release = sh(returnStdout: true, script: 'gh api ${GH_RELEASES_API_URI} | jq -e -r \'. | max_by(.id) | select(.draft == true and .name == "next").id\'').trim()
```

**Review by dduportal — APPROVED** [review]

**Comment by jglick** [comment]
I wonder if this also affects `jenkins-maven-cd-action`.

**Comment by dduportal** [comment]
> I wonder if this also affects `jenkins-maven-cd-action`.

Interesting: is your comment about the fact that `jenkins-maven-cd-action` **could** use this library, or is it about the root problem (handling presence of multiple `next` releases)?

**Comment by jglick** [comment]
I am guessing that the code here was copied from https://github.com/jenkins-infra/jenkins-maven-cd-action/blob/a0a2c865bf13c0937b93d3190ed128af1bab2947/run.sh#L11 so I was wondering whether an analogous problem affected CD workflows. Do not recall offhand. CC @timja

**Comment by dduportal** [comment]
As far as I remember, this code was written by @garethjevans , so yep, smells like same source 🤗

**Pull request merged**